### ISSUE TITLE
panic using prepared statements

### ISSUE THREAD

**Issue by dweitzman** [vitessio/vitess#5204]
We turned on prepared statements in a development environment and started getting panics:

```
server.go:269] mysql_server caught panic:
runtime error: invalid memory address or nil pointer dereference
/usr/local/go/src/runtime/panic.go:513 (0x42dba8)
/usr/local/go/src/runtime/panic.go:82 (0x42ccfd)
/usr/local/go/src/runtime/signal_unix.go:390 (0x443b11)
/vt/src/vitess.io/vitess/go/vt/vtgate/executor.go:1534 (0xac4d61)
/vt/src/vitess.io/vitess/go/vt/vtgate/executor.go:1491 (0xac4477)
/vt/src/vitess.io/vitess/go/vt/vtgate/executor.go:1443 (0xac402e)
/vt/src/vitess.io/vitess/go/vt/vtgate/vtgate.go:870 (0xadf939)
/vt/src/vitess.io/vitess/go/vt/vtgate/plugin_mysql_server.go:311 (0xac6bc5)
/vt/src/vitess.io/vitess/go/mysql/conn.go:875 (0x83cf4d)
/vt/src/vitess.io/vitess/go/mysql/server.go:457 (0x8563c0)
/usr/local/go/src/runtime/asm_amd64.s:1333 (0x45dde0)
```

**Comment by dweitzman** [comment]
This was while connecting with connector-j 5.1.47

**Comment by morgo** [comment]
PTAL @saifalharthi

**Comment by dweitzman** [comment]
Looks like that's the last line of `handlePrepare`, `return qr.Fields, err`

The implication would be that `qr` is nil

**Comment by dweitzman** [comment]
`qr` could definitely be `nil` there in the case of an error getting fields.

**Comment by saifalharthi** [comment]
@dweitzman can you share the queries the caused it?

**Comment by saifalharthi** [comment]
@dweitzman This PR should address this issue. https://github.com/vitessio/vitess/pull/5209

**Comment by morgo** [comment]
This has been fixed in #5204
